### DISCUSSION TITLE
Welcome to Caly Pdf Reader Discussions!

### DISCUSSION THREAD

**Discussion by BobLd** [BobLd/Caly#5] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

**Comment by zkvii** [comment]
I am very interested in Caly, but it would be even better if there is a WinUI branch.

**Reply by BobLd** [reply]
@zkvii thanks a lot for your interest, at the moment there's no plan for a WinUI version. May I ask why you'd like WinUI instead of Avalonia?

**Reply by zkvii** [reply]
For a PDF reader, Avalonia might already be powerful enough. However, I have the following considerations:

- Its visual effects can't be compared to WinUI.

- Additionally, whether it's Rider or Visual Studio, the error output positioning is frustrating. I don't understand why it can't directly pinpoint the error location.

Avalonia seems simpler and easier to learn compared to WinUI, but it appears that many of Caly's current bugs are related to Avalonia controls rather than rendering.skia. If I were to develop Caly using WinUI, I might use win2D/Direct2D as the rendering engine. However, these are just some personal considerations. Of course, the main challenge of a PDF reader lies in parsing, not in the UI framework.

**Comment by Leftyx** [comment]
I am also interested in the project but do not quite understand the scope.
Could you provide a bit more background in the description ?

**Reply by BobLd** [reply]
@Leftyx thanks! I agree the description needs to be improved. 

The aim of the repo is to develop a C# cross-platform open source pdf reader/viewer application, with the following abilities:
- View pdf document
- Open many documents with a tab system
- Navigate and zoom document (go to page, zoom in/out, etc.)
- Select text and copy to clipboard
- Search text

Other plans for later:
- OCR
- Table extraction

Pdf document editing is not planned for the moment, but could be added later on.

The application uses Avalonia as the UI framework. The pdf rendering to image is done with PdfPig.Rendering.Skia, which is another project I started.

You should be able to clone and build the repo on your machine, let me know if you have difficulties with that

I hope that helps!

**Reply by Leftyx** [reply]
Thanks for the swift reply and explanation.
I might be able to help a bit with the OCR as I have invested a fair amount of time recently working on it.

**Reply by BobLd** [reply]
@Leftyx I've created a new repos for the OCR processing here https://github.com/BobLd/RapidOcrNet

I'm planning to release a nuget package shortly and try integrate it to Caly. Feel free to have a look and give feedback!

**Reply by Leftyx** [reply]
> @Leftyx I've created a new repos for the OCR processing here https://github.com/BobLd/RapidOcrNet
> 
> I'm planning to release a nuget package shortly and try integrate it to Caly. Feel free to have a look and give feedback!

Hello @BobLd, thanks for the info. The repo appeared this morning on my board and I immediately tested it 😄 I must say you did a great job there.
The OCR does work but on some documents I have tested, the result are even worse than Tesseract, I Need to figure out why. 
I'll write some comments there.

**Reply by BobLd** [reply]
@Leftyx thanks a lot for the feedback. I'm surprised it's not at least as good as tesseract tbh. Hopefully this is due to pre-processing and can be fixed.

I have not really looked at the Python version of RapidOCR, but maybe it's better than their C# one. I'll try to compare both versions. Feel free to open an issue / discussion in the repos
